datastore/redis.go

```diff
@@ -40,6 +40,11 @@ var (
 	RedisBlockBuilderStatusBlacklisted BlockBuilderStatus = "blacklisted"
 )
 
+type PubKeyStatusPair struct {
```

Comment:
How about using a map instead of a custom struct? I don't have strong opinions, but currently for such use-cases maps are used across the codebase. If you strongly prefer the struct I'm fine with it too.

Reply:
changed to map